sequencer/finalizer.go

```diff
@@ -81,52 +108,105 @@ func (f *finalizer) Start() {
 				result := f.executor.ExecuteTransaction(processBatchRequest)
+
 				// executionResult := execute tx (only passing to the executor this tx, starting at currentBatch.intermediaryRoot)
-				// if ko:
-				// // decide if we [MoveTxToNotReady, DeleteTx, UpdateTx]
-				// if ok:
-				// // currentBatch.remainingResources.sub(executionResult.Counters)
-				// // if resourced overflows:
-				// // // f.worker.UpdateTx()
-				// // // continue (TODO: actually no, because we need to check the timeouts)
-				// // add tx to current batch
-				// // send tx to the DB (already finalized) through channel (async store)
-				// // remove tx from pool (already finalized) through channel (async store)
-				// // f.worker.UpdateAfterSingleSuccessfulTxExecution()
+				if result.Error != nil {
+					// // decide if we [MoveTxToNotReady, DeleteTx, UpdateTx]
+
+				} else {
+					processedTx := result.Responses[0]
+					from, err := types.Sender(types.NewEIP155Signer(processedTx.Tx.ChainId()), &processedTx.Tx)
+					if err != nil {
+						from, err = types.Sender(types.HomesteadSigner{}, &processedTx.Tx)
+					}
+
+					err = wipBatch.remainingResources.Sub(RemainingResources{
+						remainingZKCounters: result.UsedZkCounters,
+						remainingBytes:      uint64(processedTx.Tx.Size()),
+						remainingGas:        processedTx.GasUsed,
+					})
+					if err != nil {
+						f.worker.UpdateTx(processedTx.TxHash, from, wipBatch.remainingResources.remainingZKCounters)
+						// (TODO: check the timeouts)
+						continue
+					}
+
+					// We have a successful processing if we are here
+					wipBatch.intermediaryStateRoot = result.NewStateRoot
+					processBatchRequest.StateRoot = result.NewStateRoot
+					processBatchRequest.OldAccInputHash = result.NewAccInputHash
+					// We store the processed transaction, add it to the batch and delete from the pool atomically.
+					go f.dbManager.StoreProcessedTxAndDeleteFromPool(ctx, wipBatch.batchNumber, processedTx)
+					go f.worker.UpdateAfterSingleSuccessfulTxExecution(from, result.TouchedAddresses)
+				}
 			} else {
-				// check if the currentBatch is above a defined limit (almost full)
-				// if yes:
-				// // close batch and send batch info to the DB (already finalized) through channel (async store)
+				// TODO: check if the currentBatch is above a defined limit (almost full)
+				// TODO: if yes:
+				go func() {
+					ethTxs := make([]types.Transaction, 0, len(wipBatch.txs))
+					for _, txTracker := range wipBatch.txs {
+						tx, err := state.DecodeTx(string(txTracker.RawTx))
+						if err != nil {
+							log.Fatalf("failed to decode tx")
+						}
+						ethTxs = append(ethTxs, *tx)
+					}
+					receipt := state.ProcessingReceipt{
+						BatchNumber:   wipBatch.batchNumber,
+						AccInputHash:  processBatchRequest.OldAccInputHash,
+						StateRoot:     wipBatch.intermediaryStateRoot,
+						LocalExitRoot: processBatchRequest.GlobalExitRoot,
+						Txs:           ethTxs,
```

Comment:
I would like to review this to see if we can directly pass `wipBatch.txs` here. Also it would be nice if we can stop working with `types.Transaction` in the first place and use the `TxTracker` for the sequencer and "raw txs" (just the encoded tx as `[]byte`) in the state.

I'm not sure about how feasible is this, but at least is worth exploring

Reply:
Okay, will check if possible to use everywhere the raw tx.